src/main/java/seedu/address/model/person/Telegram.java

```diff
@@ -0,0 +1,85 @@
+package seedu.address.model.person;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+import java.util.Optional;
+
+/**
+ * Represents a Person's Telegram handle in UniGenda.
+ * Guarantees: immutable; is valid as declared in
+ */
+public class Telegram {
+    public static final String MESSAGE_CONSTRAINTS = "Telegram username should "
+            + "adhere to the following constraints:\n"
+            + "1. The username should only contain alphanumeric characters and underscore (_).\n"
+            + "2. The username must contain at least 5 characters.";
+    public static final String VALIDATION_REGEX = "^[a-zA-z0-9-_]{5,}$";
+    public static final Telegram EMPTY_TELEGRAM = new Telegram();
+
+    public final String value;
+
+    /**
+     * Constructs on {@code Telegram}.
+     *
+     * @param username A valid Telegram username.
+     */
+    public Telegram(String username) {
+        requireNonNull(username);
+        checkArgument(isValidUsername(username), MESSAGE_CONSTRAINTS);
+        value = username;
+    }
+
+    private Telegram() {
+        value = null;
+    }
+
+    /**
+     * Returns true if a given string is a valid Telegram username.
+     */
+    public static boolean isValidUsername(String test) {
+        return test.matches(VALIDATION_REGEX);
+    }
+
+    /**
+     * Returns true if a given Telegram is a valid Telegram object.
+     */
+    public static boolean isValidTelegram(Telegram telegram) {
+        return telegram.isEmpty() || isValidUsername(telegram.value);
+    }
+
+    /**
+     * Returns true if the Telegram object is the default Telegram.
+     */
+    public boolean isEmpty() {
+        return this.equals(EMPTY_TELEGRAM);
+    }
+
+    public Optional<Telegram> get() {
```

Comment:
I think we can reference Email and Address for this part. There is no get() method. Instead, when the other classes want to "get", they use the value attribute. Personally, I don't think Optional is should be used here. We should be using Optional in Person instead.